packages/storage/amplify_storage_s3_dart/lib/src/prefix_resolver/pass_through_prefix_resolver.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+
+/// A S3 prefix resolver that doesn't add extra prefix.
+class PassThroughPrefixResolver implements S3PrefixResolver {
+  @override
```

Comment:
+add docs w/ template/macro

```suggestion
class PassThroughPrefixResolver implements S3PrefixResolver {
  const PassThroughPrefixResolver();

  @override
```